sdcm/nemesis.py

```diff
@@ -1023,82 +1047,75 @@ def _choose_file_for_destroy(self, ks_cfs):
-            return all_files_to_destroy
-        ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_empty_tables=False)
-        if not ks_cfs:
-            raise UnsupportedNemesis(
-                'Non-system keyspace and table are not found. CorruptThenRepair nemesis can\'t be run')
+    def get_all_sstables(self, tables: list[str], node: BaseNode = None):
+        """
+        :param tables: list of tables. Format of name: <keyspace_name.table_name>
+        :param node: get SStables from the node
+        """
+        node = node or self.target_node
 
-        tables = []
-        errors = []
-        with self.cluster.cql_connection_patient(self.target_node) as session:
-            for table in ks_cfs:
-                has_data, error = self.cluster.is_table_has_data(session=session, table_name=table)
-                if has_data:
-                    tables.append(table)
-                if error:
-                    errors.append(error)
-                if len(tables) > 20:
-                    break
+        sstables = []
+        for ks_cf in tables:
+            sstable_util = SstableUtils(db_node=node, ks_cf=ks_cf)
+            ks_cf_sstables = sstable_util.get_sstables()
+            sstables.extend(ks_cf_sstables)
+            self.log.info("Sstables for % ks_cf are: %s", ks_cf, ks_cf_sstables)
+
+        self.log.info("All Sstables are: %s", sstables)
+
+        return sstables
 
+    @decorate_with_context(ignore_ycsb_connection_refused)
+    def _destroy_data_and_restart_scylla(self, keyspaces_for_destroy: list = None, sstables_to_destroy_perc: int = 50):  # pylint: disable=too-many-statements
+        tables = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_empty_tables=False,
+                                                        filter_by_keyspace=keyspaces_for_destroy)
         if not tables:
-            raise ValueError(
-                f'A non-empty user table is not found. Nemesis can\'t run. Got errors of: {errors}')
+            raise UnsupportedNemesis(
+                'Non-system keyspace and table are not found. The nemesis can\'t be run')
+
+        self.log.debug("Chosen tables: %s", tables)
 
         # Stop scylla service before deleting sstables to avoid partial deletion of files that are under compaction
         self.target_node.stop_scylla_server(verify_up=False, verify_down=True)
 
         try:
-            # Remove 5 data files
-            all_files_to_destroy = find_five_files_to_destroy()
-            if len(all_files_to_destroy) < 5:
-                self.log.warning("Expected to find 5 files to destroy, but only found %s", len(all_files_to_destroy))
-            for file_for_destroy in all_files_to_destroy:
-                result = self.target_node.remoter.sudo('rm -f %s' % file_for_destroy)
+            # Remove data files
+            if not (all_files_to_destroy := self.get_all_sstables(tables=tables, node=self.target_node)):
```

Comment:
i think if no table found it should be an error and shouldn't be skipped.

Reply:
Do you mean sstables?